File: GenshinTool.Common/Models/Domain/Concretes/StatModel.cs
using GenshinTool.Common.Models.Core.Model;
using GenshinTool.Common.Models.Domain.Interfaces;

namespace GenshinTool.Common.Models.Domain.Concretes;

public class StatModel : BaseModel, IStatModel
{
    public decimal Value { get; set; }
    public long StatNameId { get; set; }
    public IStatNameModel? StatName { get; set; }
    public bool IsMain { get; set; }
}

File: GenshinTool-SPA/src/app/Genshin/models/Stat/stat.model.ts
import { StatName } from "./statName.model";

export class Stat{
    id!: Number;
    value!: Number;
    statName!: StatName;
    isMain!: boolean;
    associationId!: Number;

    constructor(value: Number, statName: StatName, isMain: boolean = false, associationId: Number = 0) {
        this.value = value;
        this.statName = statName;
        this.isMain = isMain;
        this.associationId = associationId;
    }
}